### ISSUE TITLE
Stable Refussor - How to handle Copyright ?

### ISSUE THREAD

**Issue by MentalGear** [Stability-AI/stablediffusion#159]
As it has been [recently proven](
https://techxplore.com/news/2023-02-ai-based-image-generation-generate-trainer.html), SD outputs on occasion images extremely close to its training data.

The most profound solution of course would be to have a training set that doesn't feature any copyrighted data. Since this however, considering  costs of training, is probably not going to occur very soon for SD, I was wondering what other techniques there are to detect if the output is too close to an actual training data image?

Is there a technique that could be added to a next minor version release which will indicate when such a similarity crosses a certain `threshold` - something like a `Stable Refussor` parameter that regenerates if the output is too close?

https://techxplore.com/news/2023-02-ai-based-image-generation-generate-trainer.html

![image](https://user-images.githubusercontent.com/2837147/216581596-baa6e005-4795-4b40-9b59-af69fcb9fe8a.png)

**Comment by 5-30** [comment]
Yes, if you train a model a lot on only a few images, it will, at some point, be able to recreate something similar. However, it cannot be replicated on larger scales, like with the actual stable diffusion model, as each image has such little impact on the final output.

This was done by training a model a lot on only thousand images. On the other hand, Stable Diffusion is trained on 5 billions of images, therefore making it unable to recreate or even remember about a precise picture.
